### ISSUE TITLE
Exception on autocomplete attempt for recalled `Add-Type` cmdlet

### ISSUE THREAD

**Issue by mavaddat** [PowerShell/PSReadLine#3073]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PSReadLine)
- [X] Search the existing issues, especially the pinned issues.

### Exception report

```console
### Environment
PSReadLine: 2.2.0-beta4
PowerShell: 7.2.0-rc.1
OS: Microsoft Windows 10.0.22504
BufferWidth: 120
BufferHeight: 30

Last 200 Keys:

 n t . L e n g t h ) Enter
 $ s z E x t r a c t o r Spacebar = Spacebar N e w - O b j e c t Spacebar - T y p e N a m e Spacebar S e v e n Z i p E x t r a c t o r Spacebar - A r g u m e n t L i s t Spacebar @ ( $ s e v e n Z i p S t r e a m , [ S e v e n Z i p . I n A r c h i v e F o r m a t ] : : X Z ) Enter
 UpArrow Ctrl+a RightArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow s e v e Tab RightArrow LeftArrow RightArrow RightArrow Enter
 ] : : Shift+Tab Tab x RightArrow Enter
 Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+LeftArrow Ctrl+c RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow ) Enter
 UpArrow Ctrl+LeftArrow Ctrl+a Delete $ _ Enter
 UpArrow DownArrow UpArrow UpArrow UpArrow UpArrow UpArrow UpArrow LeftArrow - v e r Tab Enter
 UpArrow LeftArrow Spacebar - w Tab

### Exception

System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension. (Parameter 'top')
Actual value was 30.
   at System.ConsolePal.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.PSConsoleReadLine.Menu.DrawMenu(Menu previousMenu, Boolean menuSelect)
   at Microsoft.PowerShell.PSConsoleReadLine.MenuCompleteImpl(Menu menu, CommandCompletion completions)
   at Microsoft.PowerShell.PSConsoleReadLine.PossibleCompletionsImpl(CommandCompletion completions, Boolean menuSelect)
   at Microsoft.PowerShell.PSConsoleReadLine.CompleteImpl(Boolean menuSelect)
   at Microsoft.PowerShell.PSConsoleReadLine.MenuComplete(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(PSKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics, CancellationToken cancellationToken, Nullable`1 lastRunStatus)
```

### Screenshot

Just before the exception:

![screenshot of powershell in windows terminal right before the exception](https://user-images.githubusercontent.com/5055400/143933821-b09c6a8d-8cc4-413f-917b-1d930fc2de7b.png)

Just after the exception:

![screenshot of powershell in windows terminal just after the exception](https://user-images.githubusercontent.com/5055400/143933945-b0a9cbe7-29c5-4198-b66b-efde8e01df3c.png)

### Environment data

|Field|Value|
|---|---|
| PS Version | 7.2.0-rc.1 |
| PS HostName | ConsoleHost (Windows Terminal) |
| PSReadLine Version | 2.2.0-beta4 |
| PSReadLine EditMode | Windows |
| OS | 10.0.22504.1000 (WinBuild.160101.0800) |
| BufferWidth | 120 |
| BufferHeight | 30 |

### Steps to reproduce

Up-arrow to recall this command from history:

```pwsh
(Get-Item (Join-Path (Split-Path (Get-Package SevenZipExtractor).Source) lib/netstandard*) |
  Sort-Object { [version] ($_.Name -replace '^netstandard') })[-1] |
    Get-ChildItem -Filter *.dll -Recurse |
      ForEach-Object { Write-Host "Adding ``$($_.Name)``"; Add-Type -LiteralPath $_.FullName -Verbose}
```

Left-arrow to place cursor just after `-` and before `Verbose` in the `-Verbose` of the (lower) innermost command then tab to autocomplete.

### Expected behavior

Show that there are switches starting with `w` for the `Add-Type` cmdlet using the autocomplete suggestions.

### Actual behavior

Exception on tab-autocomplete attempt.

**Comment by mavaddat** [comment]
Ok, as soon as I wrote that I was not able to reproduce the error, I reproduced it.

### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PSReadLine)
- [X] Search the existing issues, especially the pinned issues.

### Exception report

### Environment
```
PSReadLine: 2.2.0-beta4
PowerShell: 7.2.0-rc.1
OS: Microsoft Windows 10.0.22504
BufferWidth: 120
BufferHeight: 30
```

Last 200 Keys:

```
 Shift+LeftArrow Shift+LeftArrow S e v e Tab RightArrow Enter
 Enter
 UpArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Ctrl+RightArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow Ctrl+LeftArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+Ctrl+LeftArrow Shift+LeftArrow Shift+Ctrl+LeftArrow Shift+LeftArrow Delete Enter
 UpArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Backspace . Tab Tab Tab Tab Tab Shift+Tab Shift+Tab Shift+Tab Shift+Tab Enter
 Enter
 UpArrow Spacebar Backspace . Tab Backspace Tab Spacebar = - Backspace Backspace - A r g u m e n t L i s t Spacebar @ ( $ s e v e n Z i p S t r e a m , [ S e v e n Z i p E x t r a c t o r . S e v e n Z i p F o r m a t ] : : X Z ) Enter
 UpArrow Ctrl+a Ctrl+x UpArrow UpArrow UpArrow UpArrow UpArrow UpArrow UpArrow UpArrow LeftArrow Shift+Ctrl+LeftArrow Shift+LeftArrow Shift+LeftArrow Delete ) Ctrl+LeftArrow Ctrl+LeftArrow ( Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow LeftArrow Spacebar - f Tab Tab Backspace Tab
```

### Exception

```
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension. (Parameter 'top')
Actual value was 30.
   at System.ConsolePal.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.PSConsoleReadLine.Menu.DrawMenu(Menu previousMenu, Boolean menuSelect)
   at Microsoft.PowerShell.PSConsoleReadLine.MenuCompleteImpl(Menu menu, CommandCompletion completions)
   at Microsoft.PowerShell.PSConsoleReadLine.PossibleCompletionsImpl(CommandCompletion completions, Boolean menuSelect)
   at Microsoft.PowerShell.PSConsoleReadLine.CompleteImpl(Boolean menuSelect)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(PSKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics, CancellationToken cancellationToken, Nullable`1 lastRunStatus)
```

### Screenshot

Just before the exception:

![screenshot of powershell in windows terminal right before the exception](https://user-images.githubusercontent.com/5055400/143943222-67a1f01b-422b-4e18-a169-0fe18add192d.png)

### Environment data

|Field|Value|
|---|---|
| PS Version | 7.2.0-rc.1 |
| PS HostName | ConsoleHost (Windows Terminal) |
| PSReadLine Version | 2.2.0-beta4 |
| PSReadLine EditMode | Windows |
| OS | 10.0.22504.1000 (WinBuild.160101.0800) |
| BufferWidth | 120 |
| BufferHeight | 30 |

### Steps to reproduce

Up-arrow to recall this command from history:

```pwsh
(Get-Item (Join-Path (Split-Path (Get-Package SevenZipExtractor).Source) lib/netstandard*) |
  Sort-Object { [version] ($_.Name -replace '^netstandard') })[-1] |
    Get-ChildItem -Filter *.dll -Recurse |
      ForEach-Object { Write-Host "Adding ``$($_.Name)``"; Add-Type -LiteralPath $_.FullName -Verbose}
```

Left-arrow to place cursor just after `-` and before `Verbose` in the `-` of the (lower) innermost command and press tab to autocomplete.



### Expected behavior

Show that there are switches starting with `w` for the `Add-Type` cmdlet using the autocomplete suggestions.

### Actual behavior

Exception.

**Comment by daxian-dbw** [comment]
This is a duplicate of #2928, which was fixed by #2949. See the GIF below.
@mavaddat Can you maybe use the private build from https://ci.appveyor.com/project/PowerShell/PSReadLine/branch/master/artifacts? It contains the fix to the menu completion, which you will find useful.

![Animation](https://user-images.githubusercontent.com/127450/143958803-fcb3214a-a99a-4db4-b7b4-7891144e7b72.gif)
